java-client/src/main/java/org/scassandra/matchers/BatchExecutionMatcher.java

```diff
@@ -0,0 +1,69 @@
+package org.scassandra.matchers;
+
+import org.hamcrest.Description;
+import org.scassandra.cql.CqlType;
+import org.scassandra.http.client.BatchExecution;
+import org.scassandra.http.client.BatchQuery;
+
+import java.util.List;
+
+public class BatchExecutionMatcher extends ScassandraMatcher<List<BatchExecution>, BatchExecution> {
+
+    private final BatchExecution expectedBatchExecution;
+
+    public BatchExecutionMatcher(BatchExecution expectedBatchExecution) {
```

Comment:
given you have a factory method can you make this package private so people don't depend on the constructor? (the others should be too, was a mistake)